kedro/pipeline/node.py

```diff
@@ -21,7 +21,7 @@
     from collections.abc import Iterable
 
 
-class Node:
+class Node:  # TODO: Too many conditionals in this function, make it work first, then fix this and remove the PLR0912 ignore
```

Comment:
Is this TODO still needed?

Reply:
I wrote the TODO and then forgot about it 🤡, thanks for pointing it out.